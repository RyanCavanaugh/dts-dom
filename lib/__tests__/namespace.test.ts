import * as dom from "../index";

describe("an emitted namespace", () => {
  describe("without members", () => {
    it("matches the snapshot", () => {
      let namespace = dom.create.namespace("Namespace");

      expect(dom.emit(namespace)).toMatchSnapshot();
    });
  });
  describe("with several members", () => {
    it("matches the snapshot", () => {
      let namespace = dom.create.namespace("Namespace");

      namespace.members.push(
        dom.create.interface("interfaceMember"),
        dom.create.alias("aliasMember", dom.type.string),
        dom.create.class("ClassMember"),
        dom.create.namespace("namespaceMember"),
        dom.create.const("constMember", dom.type.number),
        dom.create.variable("varMember", dom.type.boolean),
        dom.create.function("functionMember", [], dom.type.void),
        dom.create.enum("enumMember")
      );

      expect(dom.emit(namespace)).toMatchSnapshot();
    });
  });
});
