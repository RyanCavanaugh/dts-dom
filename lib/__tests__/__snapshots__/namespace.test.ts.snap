// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`an emitted namespace with several members matches the snapshot 1`] = `
declare namespace Namespace {
    interface interfaceMember {
    }

    type aliasMember = string;

    class ClassMember {
    }

    namespace namespaceMember {
    }

    const constMember: number;

    var varMember: boolean;

    function functionMember(): void;

    enum enumMember {
    }

}


`;

exports[`an emitted namespace without members matches the snapshot 1`] = `
declare namespace Namespace {
}


`;
